src/mscorlib/src/System/Array.cs

```diff
@@ -2636,6 +2636,47 @@ public Object Clone()
         [System.Security.SecuritySafeCritical]  // auto-generated
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         public extern void Initialize();
+
+
+        private static void ThrowArrayNullException()
```

Comment:
You should look at the existing `ThrowHelper` class and use that whenever possible. It's not very useful if we keep adding such methods all other the place.